src/components/snackbar/snackbar.tsx

```diff
@@ -99,89 +103,99 @@ export class Snackbar {
-    }
+    @State()
+    private open: boolean = false;
 
-    public connectedCallback() {
-        this.initialize();
-    }
+    @State()
+    private closing: boolean = true;
 
-    public componentDidLoad() {
-        this.initialize();
-    }
+    private snackbarId: string;
 
-    private initialize() {
-        const element = this.host.shadowRoot.querySelector('.mdc-snackbar');
-        if (!element) {
-            return;
-        }
-
-        this.mdcSnackbar = new MDCSnackbar(element);
-
-        this.mdcSnackbar.listen('MDCSnackbar:closing', this.handleMdcClosing);
+    public constructor() {
+        this.snackbarId = createRandomString();
     }
 
-    public disconnectedCallback() {
-        this.mdcSnackbar.unlisten('MDCSnackbar:closing', this.handleMdcClosing);
-        this.mdcSnackbar.destroy();
+    @Listen('changeOffset')
+    protected onChangeIndex(event: CustomEvent<number>) {
+        event.stopPropagation();
+        this.offset = event.detail;
     }
 
     /**
      * Show the snackbar
      */
     @Method()
     public async show() {
-        if (this.timeout) {
-            this.mdcSnackbar.timeoutMs = this.timeout;
+        if (this.open) {
+            return;
         }
 
-        this.mdcSnackbar.labelText = this.message;
+        this.open = true;
+        this.closing = false;
+        container.add(this.host);
 
-        this.mdcSnackbar.open();
+        if (this.timeout) {
+            setTimeout(
+                this.close,
+                Math.max(this.timeout - hideAnimationDuration, 0),
+            );
+        }
     }
 
+    private close = () => {
+        if (!this.open) {
+            return false;
+        }
+
+        this.closing = true;
+
+        setTimeout(() => {
+            this.open = false;
+            container.remove(this.host);
+            this.hide.emit();
+            this.offset = 0;
+        }, hideAnimationDuration);
+    };
+
     public render() {
         return (
             <aside
-                class={`
-                    mdc-snackbar
-                    ${this.multiline ? 'mdc-snackbar--stacked' : ''}
-                `}
+                popover="manual"
                 style={{
                     '--snackbar-timeout': `${this.timeout}ms`,
+                    '--snackbar-distance-to-top-edge': `${this.offset}px`,
+                }}
+                class={{
+                    open: this.open,
+                    'is-closing': this.closing,
                 }}
+                id={this.snackbarId}
+                role={this.open ? 'status' : undefined}
+                aria-relevant={this.open ? 'additions' : undefined}
             >
-                <div
-                    class="mdc-snackbar__surface"
-                    role="status"
-                    aria-relevant="additions"
-                >
-                    <div class="mdc-snackbar__label" aria-atomic="false"></div>
+                <div class="surface" aria-atomic="false">
```

Comment:
However, I don't really understand why MDC has `aria-atomic` explicitly set to `false`. So I left it as is.

w3 says:
> The role of status also has a default aria-atomic value of true, so that updates to the container marked with a role of status will result in the AT presenting the entire contents of the container to the user, including any author-defined labels (or additional nested elements).